### ISSUE TITLE
False positive, cpp/ql/src/Security/CWE/CWE-120/BadlyBoundedWrite.ql

### ISSUE THREAD

**Issue by peternmartin** [github/codeql#13913]
This [badly bounded write](https://github.com/github/codeql/blob/main/cpp/ql/src/Security/CWE/CWE-120/BadlyBoundedWrite.ql) query flags the following code incorrectly:

```cpp
#include <stdio.h>
#include <stdlib.h>

char* dest = "a";

int main(int argc, char* argv[])
{
  const char src[] = "Testing testing 123";
  dest = malloc(sizeof(src));
  if (dest == 0)
    return EXIT_FAILURE;

  snprintf(dest, sizeof(src), "%s", src);
  fprintf(stdout, "%s\n", dest);
  return EXIT_SUCCESS;
}
```

The `snprintf` is flagged:
```
|        bw        |                                             col1                                              |
+------------------+-----------------------------------------------------------------------------------------------+
| call to snprintf | This 'call to snprintf' operation is limited to 20 bytes but the destination is only 0 bytes. |
| call to snprintf | This 'call to snprintf' operation is limited to 20 bytes but the destination is only 2 bytes. |
```

Admittedly, the code is doing something slightly strange (that is, taking a pointer that was pointing to a string constant, and repointing it at dynamically allocated memory). However, I don't think the finding is correct.

**Comment by jketema** [comment]
We've merged a fix for this https://github.com/github/codeql/pull/13929, which should be part of the CodeQL 2.14.3 release.

**Comment by jketema** [comment]
We unfortunately had to revert the fix. I'm reopening this issue.

**Comment by jketema** [comment]
This should be fixed in CodeQL 2.16.0, which should be released sometime in January 2024.
